src/PolynomialZonotope.jl

```diff
@@ -0,0 +1,118 @@
+export AbstractNonConvexSet,
+       PolynomialZonotope,
+       dim,
+       polynomial_order,
+       order
+
+"""
+    AbstractNonConvexSet
+
+Abstract type for non-convex sets.
+"""
+abstract type AbstractNonConvexSet{N} end
```

Comment:
Let us do this properly in a separate PR:
1. Move to a new file (and maybe rename to `AbstractSet` or even `Set` or `LazySet`).¹
2. Make `LazySet` inherit from this new type and rename to `AbstractConvexSet` or just `ConvexSet`.¹ This basically means changing every file, so better merge most PRs first.
3. Make `UnionSet` inherit from this new type and include it (there are already some unit tests).

Since this is a breaking change, we should probably do it *before* releasing 1.1.0.

¹This will be something we and the user will write a lot, so a short name would be preferable.

Reply:
outsourced to #185

Reply:
> Since this is a breaking change, we should probably do it before releasing 1.1.0.

2.0 iiuc since that would be an incompatible API change (https://semver.org/)

Reply:
ok i've updated to `<: LazySet{N}`, what i should have done in the 1st place, at least 'till we resolve the approach for #185.

Reply:
#185 was created for this issue (only later). You could not have done that before ;)